python/tank/authentication/session.py

```diff
@@ -0,0 +1,230 @@
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+RV Session abstraction.
+
+The main purpose of this module is to avoid having to constantly fecth either
+the RV session or read cookies from the QWebView used for the login process.
+
+Other than from hardcoding the RV Session format in one method, we have
+attempted to keep this module relatively independant.
+"""
+
+
+class Session(object):
+    """
+    Holds session information.
+
+    A session object holds information coming from both the RV current session
+    and from the QWebView used to login/renew session from our SSO enabled site.
+
+    Attributes:
+        cookies:            A string of the base64 encoded json string of raw
+                            cookies.
+        crsf_value:         A string for the crsf value.
+        csrf_key:           A string for the crsf key.
+        error:              A string which describe the error encountered.
+        host:               A string for the hostname.
+        product:            A string for the product (rv).
+        session_expitation: An int in seconds for the UTC time of expiration.
+        session_id:         A string for the session id.
+        user_id:            A string for the user id.
+    """
+
+    def __init__(self, settings={}):
+        """
+        Constructor.
+
+        Args:
+            settings (dict):    Dictionary of element to add to the settings.
+                                Non-used key/value pairs will be discarded
+                                silently.
+        """
+        self._cookies = None
+        self._csrf_key = None
+        self._csrf_value = None
+        self._error = None
+        self._host = None
+        self._product = None
+        self._session_expiration = None
+        self._session_id = None
+        self._user_id = None
+        self.merge_settings(settings)
+
+    def merge_settings(self, settings):
+        """
+        Merge new settings with existing ones.
+
+        Args:
+            settings (dict):    Dictionary of element to merge to the settings.
+                                Non-used key/value pairs will be discarded
+                                silently.
+        """
+        for key, value in settings.iteritems():
+            _key = '_%s' % key
+            if _key in vars(self):
+                setattr(self, _key, value)
+
+    def dump(self):
+        """
+        Dump dictionary of Session settings.
+
+        This function is mainly usefull for debugging purposes.
+
+        Returns:
+            A dictionary of Session settings, leaving out the undefined ones.
+        """
+        params = {}
+        for key, value in vars(self).iteritems():
+            if value is not None:
+                params[key] = value
+        return params
+
+    def assembleSession(self):
```

Comment:
Note to self: I should stop coding in 5 different languages one the same day and confuse my standards...